the-collab-lab/tcl-75-smart-shopping-list
# [ISSUE 19] Pin and place important lists on top

## Description

This code proposes a custom hook that manages and sorts lists that were marked as important

- Created `useImportance` hook to manage a list of items and their importance status.
- Added functionality to toggle a list's importance and store the state in local storage.
     I am using localStorage to locally store references to important lists - like this we don't have to make any additional calls to firebase. 
- Implemented sorting logic to display important lists at the top based on the date they were pinned.
     I wanted to try and sort items the way that we originally wanted to implement urgency of items. I read a bit more about how sorting works in JavaScript and I think this time everything works correctly, and I am happy about this. 
- Added tooltips and aria-labels to the Pin icons.

**_Would love to change UI based on feedback_**

**Still working on tests**

## Related Issue

closes #65 

## Acceptance Criteria

- [x] There is a UX option available for me to pin/unpin important lists at the top
- [x] The pinned lists are sorted in reverse add order.
- [x] This is isolated to a per user bases.
- [x] a11y checks have been done.
- [x] Update the test to ensure coverage of this new functionality.

## Type of Changes

`enhancement`

## Updates

### Before
![Screenshot 2024-10-01 163249](https://github.com/user-attachments/assets/b9afaf96-7410-4e14-b6d9-8dd754febdff)

### After
![Screenshot 2024-10-14 143336](https://github.com/user-attachments/assets/b5f687a5-1724-43a6-9d93-5d3ce5169a40)

## Testing Steps / QA Criteria
![pinlist-ezgif com-speed](https://github.com/user-attachments/assets/7a515142-19ca-45be-ae06-558087dd56bc)

1. ran npm start and signed in
2. clicked on lists to ensure they get added in the order they were pinned
